packages/react-components/react-charts-preview/library/src/components/CommonComponents/CartesianChart.tsx

```diff
@@ -146,7 +146,7 @@ export const CartesianChart: React.FunctionComponent<IModifiedCartesianChartProp
       // eslint-disable-next-line react-hooks/exhaustive-deps
       isIntegralDataset = !props.points.some((point: { y: number }) => point.y % 1 !== 0);
     }
-  }, [props, prevProps]);
+  });
```

Comment:
why remove these dependencies

Reply:
RotateXAxisLabels function was being called inside the useEffect which will be called everytime the prop changes. But createStringXAxis function which creates the string axis is called everytime the state update happens irrespective of any prop changes because it is outside useEffect.
So the issue was createStringXAxis was being called after the rotateXAxisLabels hence, the labels were getting overlapped. So for now, I have removed the dependecy array. Need to find what exactly triggers the rotate function and hence put that in the dependecy array.

Reply:
Lets take up this issue separately.